### DISCUSSION TITLE
Nested shell with cmd2

### DISCUSSION THREAD

**Discussion by pradeep17j** [python-cmd2/cmd2#1094] [Q&A]
Hello Cmd2 developers,
Thanks for a very useful module like cmd2

Trying to build an app which has multiple levels of line interpreters.

Example:
     
      Level-1 > help
             List Help for all Level-1 commands
             

     Level-1 > switch-to-level2

     Level-1: Level-2 > help
              List help for all command in Level2

    Level-1:Level-2 > exit

    Level-1 >

In the above example , the exit command did not exit the app itself but switched from Level2 to Leve-1.

I saw some code example by Calvin (https://github.com/lobocv) but it does not mention how to handle the exit .
The exit from level-2 will terminate the whole app.

Please help me with some insight on how to handle the exit from multiple levels.

**Comment by tleonhardt** [comment]
@pradeep17j In general we discourage developers from trying to create multiple levels of nested interpreters because they tend to provide a very poor customer experience.  End users get frustrated by CLIs where they go to themselves "where did I find that feature again".

Instead you can group commands into sets and you can enable and disable groups of commands based on state.  That tends to provide a better user experience.
